retail/interactive-tutorial/RetailEvents.Samples/RejoinUserEventSample.cs

```diff
@@ -0,0 +1,123 @@
+        {
+            EventType = "detail-page-view",
+            VisitorId = "test_visitor_id",
+            EventTime = DateTime.UtcNow.ToTimestamp()
+        };
+
+        userEvent.ProductDetails.Add(productDetail);
+
+        Console.WriteLine($"User Event:");
+        Console.WriteLine($"Product detail: {userEvent.ProductDetails}");
+        Console.WriteLine($"Event type: {userEvent.EventType}");
+        Console.WriteLine($"Visitor id: {userEvent.VisitorId}");
+        Console.WriteLine($"Event time: {userEvent.EventTime}");
+        Console.WriteLine();
+
+        return userEvent;
+    }
+
+    /// <summary>
+    /// Get rejoin user event request.
+    /// </summary>
+    /// <param name="defaultCatalog">The default catalog.</param>
+    /// <returns>The rejoin user events request.</returns>
+    private static RejoinUserEventsRequest GetRejoinUserEventRequest(string defaultCatalog)
+    {
+        RejoinUserEventsRequest rejoinUserEventRequest = new RejoinUserEventsRequest
+        {
+            Parent = defaultCatalog,
+            UserEventRejoinScope = RejoinUserEventsRequest.Types.UserEventRejoinScope.UnjoinedEvents
+        };
+
+        Console.WriteLine($"Rejoin user events. request:");
+        Console.WriteLine($"Parent: {rejoinUserEventRequest.Parent}");
+        Console.WriteLine($"Rejoin scope: {rejoinUserEventRequest.UserEventRejoinScope}");
+        Console.WriteLine();
+
+        return rejoinUserEventRequest;
+    }
+
+    /// <summary>
+    /// Call the Retail API to rejoin user event.
+    /// </summary>
+    /// <param name="defaultCatalog">The default catalog.</param>
+    /// <returns>The completed rejoin response.</returns>
+    public static Operation<RejoinUserEventsResponse, RejoinUserEventsMetadata> CallRejoinUserEvents(string defaultCatalog)
+    {
+        RejoinUserEventsRequest rejoinRequest = GetRejoinUserEventRequest(defaultCatalog);
+        UserEventServiceClient client = UserEventServiceClient.Create();
+        Operation<RejoinUserEventsResponse, RejoinUserEventsMetadata> rejoinResponse = client.RejoinUserEvents(rejoinRequest);
+
+        Console.WriteLine("The rejoin operation was started:");
+        Console.WriteLine(rejoinResponse.Name);
+        Console.WriteLine();
+
+        return rejoinResponse;
+    }
+}
+// [END retail_rejoin_user_event]
+
+/// <summary>
+/// The rejoin user event tutorial class.
+/// </summary>
+public static class RejoinUserEventTutorial
+{
+    [Runner.Attributes.Example]
+    public static void PerformRejoinUserEventsOperation()
+    {
+        string projectId = Environment.GetEnvironmentVariable("GOOGLE_PROJECT_ID");
+        string defaultCatalog = $"projects/{projectId}/locations/global/catalogs/default_catalog";
+
+        // To check the error handling try to pass invalid catalog:
+        // defaultCatalog = "projects/{projectId}/locations/global/catalogs/invalid_catalog";
+
+        UserEvent userEventToWrite = RejoinUserEventSample.GetUserEvent();
```

Comment:
Same comment here, if you can reusue GetUserEvent from WriteUserEventSample that'd be best, else, move the GetUserEnvent method out of the region tags as it's not used anywhere there.

Reply:
Removed it from **Purge** and **Rejoin** samples. Now, the method **GerUserEvent()** exists only in **WriteUserEvent** sample.